src/components/plots/helpers/dot-plot/getDotPlotData.js

```diff
@@ -17,29 +19,37 @@ const getClusterNames = (state) => {
   return clusterNames;
 };
 
-const fetchPlotDataWork = (
+const getDotPlotData = (
   experimentId,
   plotUuid,
-  plotType,
 ) => async (dispatch, getState) => {
-  let config = getState().componentConfig[plotUuid]?.config ?? initialPlotConfigStates[plotType];
+  const config = getState().componentConfig[plotUuid]?.config ?? initialPlotConfigStates[plotType];
   const clusterNames = getClusterNames(getState());
   const timeout = getTimeoutForWorkerTask(getState(), 'PlotData');
 
-  config = {
-    ...config,
+  const [filterGroup, filterKey] = config.selectedPoints.split('/');
+
+  const body = {
+    name: 'DotPlot',
+    useMarkerGenes: config.useMarkerGenes,
+    numberOfMarkers: config.nMarkerGenes,
+    customGenesList: config.selectedGenes,
+    groupBy: config.selectedCellSet,
+    filterBy: {
+      group: filterGroup,
+      key: filterKey || 'All',
+    },
+    // clusterNames is used for triggering a work request upon cluster name change
     clusterNames,
   };
 
   try {
-    const body = generatePlotWorkBody(plotType, config);
-
     dispatch({
       type: PLOT_DATA_LOADING,
       payload: { plotUuid },
     });
 
-    const data = await fetchWork(
+    const { data } = await fetchWork(
```

Comment:
Why is everything wrapped in `data` now?

Reply:
So that I can return the `ETag` for the work request. This is needed to pass the ETag to the GetTrajectoryGraph call.

Reply:
couldn't we regenerate the current `Etag` for embedding when `GetTrajectoryGraph` is called instead? It's something that can be inferred from what is already stored in redux, right?